firmware/mw_dual_motor_torque_ctrl/src/dual_motor_torque_ctrl.c

```diff
@@ -1171,34 +1194,44 @@ inline void generic_motor_ISR(const HAL_MtrSelect_e mtrNum)
 			speed_pu = STPOSCONV_getVelocity(st_obj[mtrNum].posConvHandle);
 		}
 		else
-		{  // the alignment procedure is in effect
-
-			// force motor angle and speed to 0
-			angle_pu[mtrNum] = _IQ(0.0);
-			speed_pu = _IQ(0.0);
+		{
+			// The alignment procedure is in effect.
+			// For the first CTRL_State_OffLine wait time, do nothing. This allows
+			// the card to initialize during startup. Then, run the calibration procedure.
+			if(gAlignCount[mtrNum]++ < gUserParams[mtrNum].ctrlWaitTime[CTRL_State_OffLine]) {
+				angle_pu[mtrNum] = _IQ(0.0);
+				speed_pu = _IQ(0.0);
+
+				gIdq_ref_pu[mtrNum].value[1] = _IQ(0.0);
+				gIdq_ref_pu[mtrNum].value[0] = _IQ(0.0);
+			} else {
+				// force motor angle and speed to 0
+				angle_pu[mtrNum] = _IQ(0.0);
+				speed_pu = _IQ(0.0);
 
-			// set D-axis current to Rs estimation current
-			gIdq_ref_pu[mtrNum].value[0] = _IQmpy(
+				// set D-axis current to Rs estimation current
+				gIdq_ref_pu[mtrNum].value[0] = _IQmpy(
 					_IQ(gUserParams[mtrNum].maxCurrent_resEst),
-			        gCurrent_A_to_pu_sf[mtrNum]);
-			// set Q-axis current to 0
-			gIdq_ref_pu[mtrNum].value[1] = _IQ(0.0);
+					gCurrent_A_to_pu_sf[mtrNum]);
+				// set Q-axis current to 0
+				gIdq_ref_pu[mtrNum].value[1] = _IQ(0.0);
 
-			// save encoder reading when forcing motor into alignment
-			if(gUserParams[mtrNum].motor_type == MOTOR_Type_Pm)
-			{
-				ENC_setZeroOffset(encHandle[mtrNum],
+				// save encoder reading when forcing motor into alignment
+				if(gUserParams[mtrNum].motor_type == MOTOR_Type_Pm)
+				{
+					ENC_setZeroOffset(encHandle[mtrNum],
 						(uint32_t)(HAL_getQepPosnMaximum(halHandleMtr[mtrNum])
 								- HAL_getQepPosnCounts(halHandleMtr[mtrNum])));
-			}
+				}
 
-			// if alignment counter exceeds threshold, exit alignment
-			if(gAlignCount[mtrNum]++
-					>= gUserParams[mtrNum].ctrlWaitTime[CTRL_State_OffLine])
-			{
-				gMotorVars[mtrNum].Flag_enableAlignment = false;
-				gAlignCount[mtrNum] = 0;
-				gIdq_ref_pu[mtrNum].value[0] = _IQ(0.0);
+				// if alignment counter exceeds threshold, exit alignment
+				if(gAlignCount[mtrNum]
+						>= 2*gUserParams[mtrNum].ctrlWaitTime[CTRL_State_OffLine])
```

Comment:
I find the double-use of `gUserParams[mtrNum].ctrlWaitTime[CTRL_State_OffLine]` a bit confusing, I think it would be better to use a separate variable for the initial waiting time above.  This would also allow to configure the two waiting times independently.